### PR TITLE
fix Issue 20383 - [REG 2.084.z] illegal conversion from int[] to ubyt…

### PR DESCRIPTION
**Pull request by WalterBright**
…e[] is accepted

### CODE REVIEW

**Comment by WalterBright** [comment]
The problem comes about because `isArrayOpValid()` returns false, and the array code falls through to the non-array code generation, which goes on to fail miserably. The fix is to issue an error for the cast operation.

**Comment by thewilsonator** [comment]
This is now green, it should probably target stable.

File: src/dmd/arrayop.d [hunk 1]
```diff
@@ -244,6 +244,22 @@ private void buildArrayOp(Scope* sc, Expression e, Objects* tiargs, Expressions*
...
+ *      tto   = type converting to
+ * Returns:
+ *      true if can be performed by _arrayOp
+ */
+bool isArrayOpImplicitCast(TypeDArray tfrom, TypeDArray tto)
+{
+    const tyf = tfrom.nextOf().toBasetype().ty;
+    const tyt = tto  .nextOf().toBasetype().ty;
+    return tyf == tyt ||
+           tyf == Tint32 && tyt == Tfloat64;
```
Comment (Geod24) [inline src/dmd/arrayop.d]: Why the special casing ?

Comment (WalterBright) [reply src/dmd/arrayop.d]: It's the only one I know of. Can always add more if they come up.

**Comment by WalterBright** [comment]
The circleci failure:
```
./.circleci/run.sh: line 116: MODEL: unbound variable
```
as I recall, this was fixed in master.

**Comment by WalterBright** [comment]
I tried `git rebase upstream/stable` but git said it was up to date.

**Comment by thewilsonator** [comment]
Maybe just force push.

**Comment by WalterBright** [comment]
Errors in buildkite seem to be due to outdated druntime and phobos in those tests, unrelated to this PR.

**Pull request merged**